### PR TITLE
[ADD] hooks: add useExternalListener hook

### LINKED ISSUE #608: add useExternalListener to owl
it's very useful

### PR DESCRIPTION
**Pull request by ged-odoo**
It is very useful.

Also, this commit prettifies the code.

closes #608

### CODE REVIEW

**Comment by ged-odoo** [comment]
oops, no test

**Comment by ged-odoo** [comment]
@aab-odoo now I added a test

File: doc/reference/hooks.md [hunk 1]
```diff
@@ -354,6 +355,17 @@ will be added to the parent environment. Note that it will extend, not replace
 the parent environment. And of course, the parent environment will not be
 affected.
 
+### `useExternalListener`
+
+The `useExternalListener` helps solve a very common problem: adding and removing
```
Comment (aab-odoo) [inline doc/reference/hooks.md]: missing hook? 

> The `useExternalListener` _hook_ helps...

File: src/hooks.ts [hunk 2]
```diff
@@ -131,3 +131,27 @@ export function useSubEnv(nextEnv) {
   const component = Component.current!;
   component.env = Object.assign(Object.create(component.env), nextEnv);
 }
+
+// -----------------------------------------------------------------------------
+// useExternalListener
+// -----------------------------------------------------------------------------
+
+/**
+ * When component needs to listen to DOM Events on element(s) that is not
```
Comment (aab-odoo) [inline src/hooks.ts]: When a component

Comment (aab-odoo) [reply src/hooks.ts]: elements that are not

**Pull request merged**